### ISSUE TITLE
Strange behavior when merging labels using label_join

### ISSUE THREAD

**Issue by n9** [VictoriaMetrics/VictoriaMetrics#5881]
### Describe the bug

MetricsQL 

`label_join(rate(container_cpu_usage_seconds_total{cpu="total",instance="foo"}), "id", " ", "namespace", "pod", "container", "id")`

returns 

```json
"metric": {
      "beta_kubernetes_io_arch": "amd64",
      "beta_kubernetes_io_os": "linux",
      "cluster": "arch-d",
      "cpu": "total",
      "id": "      bepods.slice",
      "instance": "foo",
      "job": "kubernetes-nodes-cadvisor",
      "kubernetes_io_arch": "amd64",
      "kubernetes_io_hostname": "foo",
      "kubernetes_io_os": "linux"
},
```

The value of label `id` is trimmed `/kubepods.slice` -> `bepods.slice` during join.
It seems that it only happens when joining into a label with same name as source. (For instance, when joining into `jid` instead of `id` it works properly.)

Other broken result:

```json
"metric": {
      "beta_kubernetes_io_arch": "amd64",
      "beta_kubernetes_io_os": "linux",
      "cluster": "arch-d",
      "container": "coredns",
      "cpu": "total",
      "id": "kube-system coredns-5d78c9869d-cgg2g coredns kube-system coredns-5d78c9869d-cgg2g coredns pods-burstable-pod43ee1a64_b397_43c4_8cb7_7cc095bb4f59.slice/cri-containerd-db065d66d9079a8a19d5479ff8a4a241d7d18c2dd588e3bf534cc5161b0a34f4.scope",
      "image": "registry.k8s.io/coredns/coredns:v1.10.1",
      "instance": "foo",
      "job": "kubernetes-nodes-cadvisor",
      "kubernetes_io_arch": "amd64",
      "kubernetes_io_hostname": "foo",
      "kubernetes_io_os": "linux",
      "name": "db065d66d9079a8a19d5479ff8a4a241d7d18c2dd588e3bf534cc5161b0a34f4",
      "namespace": "kube-system",
      "pod": "coredns-5d78c9869d-cgg2g"
},

```
In this case `id` contains repeated value `kube-system coredns-5d78c9869d-cgg2g coredns` and the original value of `id` is also trimmed. 

### To Reproduce

Use `label_join` to merge labels.

### Version

v1.93.12

https://registry.hub.docker.com/layers/victoriametrics/victoria-metrics/v1.93.12/images/sha256-8edb62a927a94dbdb1e1168d64fa5fa8ed677566aa6c4f17182082d6b6c4fadd?context=explore

**Comment by valyala** [comment]
FYI, this bug has been also fixed in [v1.97.3 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.3).

**Comment by valyala** [comment]
This bug has been fixed in [v1.99.0 release](https://docs.victoriametrics.com/changelog/). Closing the issue as fixed then.
